mongorestore/mongorestore_test.go

```diff
@@ -1936,5 +1950,61 @@ func TestRestoreTimeseriesCollections(t *testing.T) {
 				So(clusteredIdx, ShouldBeTrue)
 			}
 		})
+
+		Convey("system.buckets should be renamed if the timeseries collection is renamed", func() {
+			args = append(args, NSFromOption, dbName+".foo_ts", NSToOption, dbName+".foo_rename_ts", DirectoryOption, "testdata/timeseries_tests/ts_dump")
+			restore, err := getRestoreWithArgs(args...)
+			So(err, ShouldBeNil)
+
+			result := restore.Restore()
+			defer restore.Close()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 10)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := testdb.Collection("foo_ts").CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 0)
+
+			count, err = testdb.Collection("system.buckets.foo_ts").CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 0)
+
+			count, err = testdb.Collection("foo_rename_ts").CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 1000)
+
+			count, err = testdb.Collection("system.buckets.foo_rename_ts").CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 10)
+		})
+
+		Convey("system.buckets should not be renamed if the timeseries collection is not renamed", func() {
```

Comment:
Is it more accurate to say if the system bucket is rename?

Reply:
@huan-Mongo do you mean something like:

```
"system.buckets collection should not be renamed if the timeseries collection is not renamed, even if the user tries to rename the system.buckets collection"
```

Reply:
Yes, that's what I meant。

Reply:
Yes, that's what I meant.